### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by tanishk15sharma**
**AUTHENTICATION FUNCTIONILITY**

- SIGN UP USER 
- LOGIN USER
- Store user data in reducer state.

### CODE REVIEW

File: src/contexts/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { createContext, useContext, useReducer } from "react";
+import { authReducer } from "../reducers/auth-reducer";
+
+const authContext = createContext();
+
+const AuthContextProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(authReducer, {
+    isAuth: localStorage.getItem("token") ? true : false,
+    // !! bhi de sakte hai
```
Comment (SukanyaSen530) [inline src/contexts/auth-context.js:9]: Remove this

**Review by SukanyaSen530 — APPROVED** [review]

File: src/components/login/Login.js [hunk 2]
```diff
@@ -1,16 +1,51 @@
...
+
+  const inputHandler = (e) => {
+    setUserData((data) => ({ ...data, [e.target.name]: e.target.value }));
+  };
+  // console.log(userData);
+
+  const postLoginDetails = async ({ email, password }) => {
+    authDispatch({ type: "USER_LOAD" });
+    try {
+      console.log(email, password);
```
Comment (prathmeshjagtap) [inline src/components/login/Login.js:19]: remove console log

File: src/components/login/Login.js [hunk 3]
```diff
@@ -1,16 +1,51 @@
...
+      let { data } = await axios.post("/api/auth/login", { email, password });
+      authDispatch({ type: "USER_LOAD_SUCCESS", payload: data });
+      console.log(data);
+      localStorage.setItem("token", data.encodedToken);
+    } catch (err) {
+      console.log(err);
+    }
+  };
+  console.log(authState);
```
Comment (prathmeshjagtap) [inline src/components/login/Login.js:28]: remove console log

File: src/components/signup/SignUp.js [hunk 4]
```diff
@@ -1,25 +1,93 @@
...
+
+    try {
+      const { data } = await axios.post("/api/auth/signup", userInfo);
+      authDispatch({ type: "USER_LOAD_SUCCESS", payload: data.createdUser });
+      console.log(data);
+      localStorage.setItem("token", data.encodedToken);
+    } catch (err) {
+      console.log(err);
+    }
+    // console.log(authState);
```
Comment (prathmeshjagtap) [inline src/components/signup/SignUp.js:34]: you can remove this

File: src/components/signup/SignUp.js [hunk 5]
```diff
@@ -1,25 +1,93 @@
...
+  };
+
+  useEffect(() => {
+    console.log(authState);
+  });
```
Comment (prathmeshjagtap) [inline src/components/signup/SignUp.js:39]: why are you using this useEffect just to console.log ? , you can remove this

**Review by prathmeshjagtap — CHANGES_REQUESTED** [review]

File: src/components/signup/SignUp.js (hunk 5 above)
Comment (tanishk15sharma) [reply src/components/signup/SignUp.js:39]: it was just testing purpose , now I will remove it .

File: src/contexts/auth-context.js (hunk 1 above)
Comment (tanishk15sharma) [reply src/contexts/auth-context.js:9]: okay

File: src/components/signup/SignUp.js (hunk 4 above)
Comment (tanishk15sharma) [reply src/components/signup/SignUp.js:34]: yes i will remove them

**Pull request merged**